### PR TITLE
ogc api processes subscriber

### PR DESCRIPTION
**Pull request by totycro**
# Overview

 Add support for OGC API Processes Subscriber

The subscription URLs are passed to the manager, which
then has to call them appropriately.

By default, managers have the attribute `supports_subscribing`
set to `False` in order to not break the API for these. The
subscriptions are only passed to if this is set to `True`

# Related Issue / Discussion

https://github.com/geopython/pygeoapi/issues/1285

# Additional Information

# Contributions and Licensing

(as per https://github.com/geopython/pygeoapi/blob/master/CONTRIBUTING.md#contributions-and-licensing)

- [X] I'd like to contribute [feature X|bugfix Y|docs|something else] to pygeoapi. I confirm that my contributions to pygeoapi will be compatible with the pygeoapi license guidelines at the time of contribution.
- [X] I have already previously agreed to the pygeoapi Contributions and Licensing Guidelines

### CODE REVIEW

**Review by ricardogsilva — CHANGES_REQUESTED** [review]
I'm not a core committer, so take these with a grain of salt while you wait for a review from maintainers :smile: 

Nice work adding this missing functionality, I just wish we could make it a bit more generic and enforce its support for all managers.

File: pygeoapi/util.py
Comment (totycro) [reply pygeoapi/util.py]: Thanks! I wasn't sure if I read all parts of the standard correctly, so it's great that someone keeps a close eye on this :)

I've now made `success_uri mandatory.

File: pygeoapi/util.py
Comment (totycro) [reply pygeoapi/util.py:575]: I pretty much agree, I prefer a more explicit style here.

File: pygeoapi/process/manager/base.py
Comment (totycro) [reply pygeoapi/process/manager/base.py:401]: You're right, I've fixed this by providing a default value.

File: pygeoapi/process/manager/base.py
Comment (totycro) [reply pygeoapi/process/manager/base.py:340]: Now there is one :)

**Comment by totycro** [comment]
@ricardogsilva Thanks a lot for the quick feedback!

About the question whether subscriber support should be mandatory in the manager:

My reasoning was that currently existing managers wouldn't support it, but I'm not sure how many of those are really out there.

I'm currently maintaining the Kubernetes Job Manager [here](https://github.com/eurodatacube/pygeoapi-kubernetes-papermill/blob/main/pygeoapi_kubernetes_papermill/kubernetes.py#L290), which passes the execution to a Kubernetes job. So the manager itself doesn't really know when the job finishes and can't easily send success/failed notifications. I'm going to try to get this to work somehow, but in other cases maintainers might not want to add support for this.

However if there are no such managers out there, we can simply make it mandatory and remove the `supports_subscribing` logic.

File: pygeoapi/process/manager/base.py
Comment (francescoingv) [inline pygeoapi/process/manager/base.py:210]: Missing description of `subscriber` parameter.

Comment (totycro) [reply pygeoapi/process/manager/base.py:210]: I don't have the permissions to push the repo directly and not sure if this warrants a PR, would you mind adding something like this line to the sync and async execute methods?
```:param subscriber: optional `Subscriber` specifying callback URLs ```

Comment (tomkralidis) [reply pygeoapi/process/manager/base.py:210]: I'll push a fix.

**Pull request merged**
